### DISCUSSION TITLE
How to get Sensor value using script INA219

### DISCUSSION THREAD

**Discussion by fhorinek** [arendst/Tasmota#19289] [Q&A] [answered]
Hi I have connected INA219 via I2C to esp8266.
I can see the voltage, current and power in web interface and via MQTT.
Therefore I assume that it is configured properly.

However don't know access the sensor value in script.
I have tried several variant, that might work since they are used like this in rules.
The examples in documentation suggest this as well.
```
vol=INA219#Voltage
vol=INA#Voltage
vol=Ina219#Voltage
print %vol%
print %INA219#Voltage%
```
print output is `???`, tried in >F >S >T sections

My configuration is
```
#ifndef USE_INA219
#define USE_INA219    
#endif

#ifndef USE_SCRIPT

#define USE_SCRIPT
#define USE_SCRIPT_WEB_DISPLAY
#define USE_SCRIPT_SUB_COMMAND
#define USE_SCRIPT_TIMER
#define USE_ENERGY_SENSOR
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif 
```

I have latest development branch and master from github
What am I doing wrong?

**Comment by sfromis** [comment, marked as answer]
Did you make sure to have your variable created in the `>D` section?

I'm not using scripting, but the correct format for the name would be `INA219#Voltage` (to match what you see in MQTT payloads). Note that if you have more than one INA219 sensor, you should enable `SetOption64` to have the sensor name become `INA219_1` instead of `INA219-1` which is not supported in scripting.

**Reply by fhorinek** [reply]
Thank you 
I needed to define `vol` in `>D`
strange that `print %INA219#Voltage%` is not working, need to be in variable first

**Reply by sfromis** [reply]
No, that's not strange. You should not expect constructs to work outside of what's specifically documented.

**Reply by fhorinek** [reply]
True. Did not want to be ungrateful. I was just extrapolating, since it is able to access variables and arrays.
It needs to be simple because of the tiny footprint, still powerful.
It amazes me still how tasmota evolved from alternative firmware for sonoff device to this huge universal platform.

**Comment by pkkrusty** [comment]
How is ina published when teleperiod publishes? For a INA226, it looks like INA226-1. Thus I have to implement setoption64 1 to switch to underscore for the default dash, since that interferes with math in scripting. So in the >T section, I use value=INA226_1#Current

Ina219 may be similar. 

Be sure only assign variables values in >T section, no other logic.

**Reply by sfromis** [reply]
With only one INA219, the default is payloads like `"INA219":{"Id":40,"Voltage":4.353,"Current":0.009,"Power":0.039}` without the number, until adding additional sensor boards.
